pkg/controller/telemetry/metric.go

```diff
@@ -437,36 +455,29 @@ func (m *MetricController) buildWorkloadMetric(data *requestMetric) workloadMetr
 	return *trafficLabels
 }
 
-func (m *MetricController) buildServiceMetric(data *requestMetric) (serviceMetricLabels, logInfo) {
-	var dstAddr, srcAddr []byte
-	for i := range data.dst {
-		dstAddr = binary.LittleEndian.AppendUint32(dstAddr, data.dst[i])
-		srcAddr = binary.LittleEndian.AppendUint32(srcAddr, data.src[i])
+// guessWorkloadService find the first service of the workload that matches the destination port
+func (m *MetricController) guessWorkloadService(workload *workloadapi.Workload, dstIp string, targetPort uint32) *workloadapi.Service {
```

Comment:
where do you pass the dstIP?

Reply:
sorry, my bad, I forget to remove the redundant code here